packages/ui/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import {
+  forwardRef,
+  HTMLChakraProps,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  UseNumberInputProps,
+} from '@chakra-ui/react';
+
+type NumberInputSize = 'sm' | 'md' | 'lg';
+
+type NumberInputMode = 'numeric' | 'decimal';
+
+export interface NumberInputProps extends Omit<HTMLChakraProps<'div'>, keyof UseNumberInputProps> {
+  /**
+   * Determines number input's paddings and height.
+   */
+  size: NumberInputSize;
+  /**
+   * Determines type of data that might be entered by the user (numeric or decimal).
+   */
+  inputMode?: NumberInputMode;
+  /**
+   * Determines if number input is disabled or not .
+   */
+  disabled?: boolean;
+  value?: number;
+  onChange?: (arg: string) => void;
+  max?: number;
+  min?: number;
+  precision?: number;
+  step?: number;
+}
+
+export const lgStyleProps = {
+  height: '48px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const mdStyleProps = {
+  height: '40px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const smStyleProps = {
+  height: '32px',
+  padding: '1px 0px 1px 12px',
+};
```

Comment:
This should be specified on a theme level. You can find more [here](https://chakra-ui.com/docs/theming/component-style).